db/gorm.go

```diff
@@ -89,8 +113,9 @@ func (ts mappedSelector[T]) Query(ctx context.Context, clauses string, args ...i
 // the query. The caller is also responsible for ensuring that the clauses
 // argument does not contain any user-influenced input.
 func (ts mappedSelector[T]) QueryFrom(ctx context.Context, tablename string, clauses string, args ...interface{}) (Rows[T], error) {
-	if !mariaDBUnquotedIdentifierRE.MatchString(tablename) {
-		return nil, fmt.Errorf("unsafe db table name %q", tablename)
+	err := validMariaDBUnquotedIdentifier(tablename)
+	if err != nil {
+		return nil, err
```

Comment:
nit here and on other places where you handle an error from the new helper. See also my other comment above:
```suggestion
		return nil, fmt.Errorf("table name: %w", err)
```